Add example to show splitview for iPads

ashleyng opened the pull request:
<!--
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.
-->

What does this implement/fix? Explain your changes.
---------------------------------------------------
Show an example of having an iMessage style split view. 
Reuses `BasicExampleViewController` to create the messaging interface and follows [InputBarAccessoryView](https://github.com/nathantannar4/InputBarAccessoryView) subview [example](https://github.com/nathantannar4/InputBarAccessoryView/blob/master/Example/Example/SubviewExampleViewController.swift) to actually setup the iMessage-esque layout
👻

![iPad_portrait_no_selection](https://user-images.githubusercontent.com/5355231/76694133-01f39400-663d-11ea-9105-69446315b0fe.png)
![iPad_landscape_subview_selected](https://user-images.githubusercontent.com/5355231/76694134-0324c100-663d-11ea-9032-577881dbfeac.png)
![iPad_landscape_keyboard](https://user-images.githubusercontent.com/5355231/76694147-2a7b8e00-663d-11ea-8561-3a2458ca7371.png)
![iPhoneXr_subview_selected](https://user-images.githubusercontent.com/5355231/76694135-04ee8480-663d-11ea-8801-b668b1d3edd5.png)



Does this close any currently open issues?
------------------------------------------
The issue is already closed but is related to:
https://github.com/MessageKit/MessageKit/issues/465


Any relevant logs, error output, etc?
-------------------------------------
<!--
If the logs is quite long, please paste to https://ghostbin.com/ and insert the link here.
-->

Any other comments?
-------------------
-  had to use the "last resort" bottom inset in `MessageSubviewViewController` but I figure that fix should be in a different PR
-  had to disable a swiftlint rule for cyclomatic complexity in `LaunchViewController`, but I don't think there's really a good way to split up the switch statement. As well as it's only in the example app.

Where has this been tested?
---------------------------
**Devices/Simulators:** Simulator  iPad Pro (11-inch) iPhone 11 pro, iPhone Xr

**iOS Version:** iOS 13.3, Xr was on 12.4

**Swift Version:** Swift 5 

**MessageKit Version:** 3.1.0-beta.1

ashleyng committed:
Add support to show splitview for iPads

Alexander-Frost commented:
This looks really cool!

Kaspik committed:
Merge branch 'development' into Support_iPad_SplitView

ashleyng committed:
Merge branch 'development' into Support_iPad_SplitView

Kaspik committed:
Merge branch 'development' into Support_iPad_SplitView

Kaspik commented on Example/Sources/AppDelegate.swift:
```diff
@@ -30,9 +30,15 @@ final internal class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+
+        let masterVC = NavigationController(rootViewController: LaunchViewController())
```
I believe example app uses full naming (aka `ViewController`, not `VC`), could we stick with that please?

Kaspik commented on Example/Sources/View Controllers/LaunchViewController.swift:
```diff
@@ -69,19 +69,34 @@ final internal class LaunchViewController: UITableViewController {
         switch cell {
         case "Basic Example":
-            navigationController?.pushViewController(BasicExampleViewController(), animated: true)
+            let vc = BasicExampleViewController()
```
^^

Kaspik commented on Example/Sources/View Controllers/MessageSubviewContainerViewController.swift:
```diff
@@ -0,0 +1,41 @@
+//
+//  MessageSubviewContainerViewController.swift
```
This needs proper MIT License Header, copy that from `AppDelegate.swift` for example please :)

Kaspik commented on Example/Sources/View Controllers/MessageSubviewViewController.swift:
```diff
@@ -0,0 +1,33 @@
+//
+//  MessageSubviewViewController.swift
```
This needs proper MIT License Header, copy that from `AppDelegate.swift` for example please :)

Kaspik commented on Example/Sources/View Controllers/MessageSubviewViewController.swift:
```diff
@@ -0,0 +1,33 @@
+import Foundation
+import InputBarAccessoryView
+
+class MessageSubviewViewController: BasicExampleViewController {
```
This can be final one?

Kaspik commented on Example/Sources/View Controllers/MessageSubviewViewController.swift:
```diff
@@ -0,0 +1,33 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        subviewInputBar.delegate = self
+        additionalBottomInset = 88
```
Would be nice to add some docu / info - why 88? (for future usage/readers)

Kaspik reviewed (changes requested):
Couple of small things only, thanks for this!

ashleyng committed:
Address code review

- change varibale names with VC to ViewController
- add MIT License Header to new files
- add documentation about using a height constant

kinoroy committed:
Merge branch 'development' into Support_iPad_SplitView

kinoroy commented:
Gonna merge this as @Kaspik 's requested changes were satisfied in d2a34894a5008e85cca1cab5367fa04f19bcc39d.

SD10 commented:
Thank you for contributing to MessageKit! I've invited you to join the MessageKit GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/MessageKit/MessageKit/blob/master/CONTRIBUTING.md) and join the [MessageKit Slack channel](https://join.slack.com/t/messagekit/shared_invite/enQtMjI4NzIzNzMyMzU0LTEyNmYzNzBlNjNiYjAwYjY1ZjhkMmVjNDE4OGNlMDY1OTRmNzVmMzI4MTg3OWQzZTM2NGU4NTU3ZDE5NzkwNzQ). Feel free to reach out if you have any questions! :smiley:

isupremedyou commented on Example/Sources/View Controllers/ChatViewController.swift:
```diff
@@ -313,18 +319,19 @@ extension ChatViewController: InputBarAccessoryViewDelegate {
-        messageInputBar.inputTextView.placeholder = "Sending..."
+        inputBar.sendButton.startAnimating()
+        inputBar.inputTextView.placeholder = "Sending..."
+        // Resign first responder for iPad split view
```
@ashleyng I was wondering what the purpose behind resigning the first responder is. What was the cause for adding this to the example? I am currently working through an iPad Split View implementation and noticed your comment in the example.

Thanks!

Rspoon3 replied:
I would also like to know this.
